src/PreviewCanvas/Controls.js

```diff
@@ -28,6 +30,21 @@ const Controls = (props) => {
     setIsPlaying(false);
   };
 
+  useEffect(() => {
+    const isVideoFinished = () => {
+      if (videoContext.currentTime >= videoContext.duration) {
+        setIsPlaying(false)
+      }
+        requestAnimationFrame(isVideoFinished)
+      }
+
+    setVideoContext(props.videoContext);
```

Comment:
This will trigger the useEffect again, perhaps set this when you initially receive the props?
```suggestion
```